File: backend/models/hospital.py
from utils.db import db
from datetime import datetime

class Hospital(db.Model):
    __tablename__ = 'hospitals'
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id', ondelete="CASCADE"), nullable=False)
    name = db.Column(db.String(150), nullable=False)
    location = db.Column(db.String(150), nullable=False)
    certification_status = db.Column(db.Enum('Pending', 'Certified', 'Rejected', name='hospital_certification'), default='Pending')
    is_paid = db.Column(db.Boolean, default=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    user = db.relationship('User', backref=db.backref('hospital_profile', uselist=False, cascade="all, delete-orphan"))

    def to_dict(self):
        return {
            'id': self.id,
            'user_id': self.user_id,
            'name': self.name,
            'location': self.location,
            'certification_status': self.certification_status,
            'is_paid': self.is_paid,
            'created_at': self.created_at.isoformat()
        }

File: backend-node/models/Hospital.js
const mongoose = require('mongoose');

const hospitalSchema = new mongoose.Schema({
  user_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  name: { type: String, required: true },
  location: { type: String, required: true },
  certification_status: { 
    type: String, 
    enum: ['Pending', 'Certified', 'Rejected'], 
    default: 'Pending' 
  },
  is_paid: { type: Boolean, default: false },
  created_at: { type: Date, default: Date.now }
});

hospitalSchema.set('toJSON', {
  transform: (document, returnedObject) => {
    returnedObject.id = returnedObject._id.toString();
    returnedObject.user_id = returnedObject.user_id ? returnedObject.user_id.toString() : null;
    delete returnedObject._id;
    delete returnedObject.__v;
  }
});

module.exports = mongoose.model('Hospital', hospitalSchema);
